src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -147,6 +149,33 @@ private void updateActionForDeployment(Map<String, Object> newConfig, Deployment
             return;
         }
 
+        // As long as the timeout is not 0, we will try to run the connectivity validation
+        Integer timeout = deployment.getDeploymentDocumentObj().getConfigurationValidationPolicy().timeoutInSeconds();
+        if (timeout != null && !timeout.equals(0)) {
+            try (ConnectivityValidator connectivityValidator = new ConnectivityValidator(
+                    deviceConfiguration.getConfigurationPOJO(),
+                    newConfig,
+                    mqttClient
+            )) {
+                boolean needsValidation = connectivityValidator.mqttClientNeedsValidation(deviceConfiguration);
+                if (needsValidation && !connectivityValidator.mqttClientCanConnect(totallyCompleteFuture)) {
+                    return;
+                }
+
+                needsValidation = connectivityValidator.httpClientNeedsValidation(deviceConfiguration);
+                if (needsValidation && !connectivityValidator.httpClientCanConnect(totallyCompleteFuture)) {
+                    return;
+                }
```

Comment:
I think this should be a separate function in ConnectivityValidator

Reply:
Do you mean moving the code block into a funciton inside of ConnectivityValidator, something like:

 ```
try (ConnectivityValidator connectivityValidator = new ConnectivityValidator(
                    deviceConfiguration.getConfigurationPOJO(),
                    newConfig,
                    mqttClient
            )) {
                if (!connectivityValidator.validateConnectivity(totallyCompleteFuture, deviceConfiguration)) {
                    return;
                }
}
```

Reply:
Also what if the validation takes longer than the timeout?

Reply:
We use the timeout only to determine whether or not to run the validation logic.